bytecomp/matching.ml

```diff
@@ -3027,41 +3521,52 @@ let rec map_return f = function
    can be costly (one unnecessary tuple allocation).
 *)
 
-let assign_pat opt nraise catch_ids loc pat lam =
-  let rec collect acc pat lam = match pat.pat_desc, lam with
-  | Tpat_tuple patl, Lprim(Pmakeblock _, lams, _) ->
+let assign_pat opt nraise catch_ids loc pat action_loc lam =
```

Comment:
If I understand correctly, `action_loc` and `lam` should be a single `Lambda.block` value.

Reply:
This has all been fixed, but I'm unsure that I've actually got the semantics of locations correct here.  Perhaps @trefis can check when he completes his review.